### PR TITLE
Remove extraneous whitespace

### PR DESCRIPTION
**Pull request by SunriseLong**
This is a:
- [x ] bug fix PR with no breaking changes — please ensure the base branch is `main`
- [ ] new functionality — please ensure the base branch is the latest `dev/` branch
- [ ] a breaking change — please ensure the base branch is the latest `dev/` branch

## Description & motivation

This PR addresses some of https://github.com/dbt-labs/dbt-utils/issues/513 for the issues affect my team while linting dbt with SQLFluff. Since some dbt_util macros do not have whitespace control, it leads to rule [L039 ](https://docs.sqlfluff.com/en/stable/rules.html#sqlfluff.core.rules.Rule_L039)failing. 

I have only tested on Redshift since this is why my team uses. Happy to apply to other adapters, just let me know what to do since I'll be flying blind. 

Macros updated:
*  date_trunc
```
-- raw
SELECT
    {{ dbt_utils.date_trunc('day', 'work_orders.start_time') }}::DATE AS start_time
FROM work_orders

-- before
SELECT

    date_trunc('day', dispatch.dispatch_request_started_at)
::DATE AS day,
FROM work_orders

-- after
SELECT
    date_trunc('day', dispatch.dispatch_request_started_at)::DATE AS day,
FROM work_orders
``` 

* datediff
```
-- raw 
SELECT
    {{ dbt_utils.datediff(date_1, date_2, 'day') }} AS day
FROM work_orders

-- before
SELECT
    

    datediff(
        day,
        date_1,
        date_2
        )

 AS day
FROM work_orders

-- after
SELECT
    datediff(
        day,
        date_1,
        date_2
        ) AS day
FROM work_orders
```

## Checklist
- [x] I have verified that these changes work locally on the following warehouses (Note: it's okay if you do not have access to all warehouses, this helps us understand what has been covered)
    - [ ] BigQuery
    - [ ] Postgres
    - [x] Redshift
    - [ ] Snowflake
- [x] I followed guidelines to ensure that my changes will work on "non-core" adapters by:
    - [ ] dispatching any new macro(s) so non-core adapters can also use them (e.g. [the `star()` source](https://github.com/fishtown-analytics/dbt-utils/blob/master/macros/sql/star.sql))
    - [ ] using the `limit_zero()` macro in place of the literal string: `limit 0`
    - [ ] using `dbt_utils.type_*` macros instead of explicit datatypes (e.g. `dbt_utils.type_timestamp()` instead of `TIMESTAMP`
- [ ] I have updated the README.md (if applicable)
- [ ] I have added tests & descriptions to my models (and macros if applicable)
- [x] I have added an entry to CHANGELOG.md

### CODE REVIEW

File: macros/cross_db_utils/datediff.sql [hunk 1]
```diff
@@ -3,15 +3,15 @@
 {% endmacro %}
 
 
-{% macro default__datediff(first_date, second_date, datepart) %}
+{%- macro default__datediff(first_date, second_date, datepart) -%}
```
Comment (joellabes) [inline macros/cross_db_utils/datediff.sql:6]: The leading `-` doesn't do anything (there is no whitespace prior to it to remove) - this is also true for the trailing `-` in the `endmacro` block, but I haven't added this suggestion in every place
```suggestion
{% macro default__datediff(first_date, second_date, datepart) -%}
```

File: macros/cross_db_utils/date_trunc.sql [hunk 2]
```diff
@@ -2,9 +2,9 @@
   {{ return(adapter.dispatch('date_trunc', 'dbt_utils') (datepart, date)) }}
 {%- endmacro %}
 
-{% macro default__date_trunc(datepart, date) %}
+{%- macro default__date_trunc(datepart, date) -%}
     date_trunc('{{datepart}}', {{date}})
-{% endmacro %}
+{%- endmacro -%}
 
 {% macro bigquery__date_trunc(datepart, date) %}
```
Comment (joellabes) [inline macros/cross_db_utils/date_trunc.sql:9]: For cross-db support, you'll need to do the same thing on each of the adapter-specific macros, like this: 

```suggestion
{% macro bigquery__date_trunc(datepart, date) -%}
```

Again, I haven't applied this _everywhere_ it's necessary - just one example

**Review by joellabes — CHANGES_REQUESTED** [review]
Hi @SunriseLong, thank you for doing this! These look good, a couple of nitpicks but I'm very excited to get this in! Thanks for providing the before/after examples, made it super easy to see what had changed 🙏

**Comment by SunriseLong** [comment]
Thanks addressed. The `redshift__datediff()` whitespace control is a little redundant since it returns `default__datediff()` which is already controlled.

Haven't tested the gcp adapters btw

**Review by joellabes — DISMISSED** [review]

**Comment by joellabes** [comment]
This is looking great! The integration tests exercise BigQuery (as well as Snowflake, Redshift and Postgres) so we should be OK on GCP 👍 

Going to change the base branch to `next/patch` which might cause some chaos, but otherwise this is good to merge! Thanks for your contribution 💟

**Review by joellabes — APPROVED** [review]

**Pull request merged**